core/src/test/java/tachyon/client/RemoteBlockInStreamTest.java

```diff
@@ -15,43 +15,72 @@
 package tachyon.client;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 
-import org.junit.AfterClass;
+import org.junit.After;
 import org.junit.Assert;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import tachyon.TachyonURI;
 import tachyon.TestUtils;
+import tachyon.conf.CommonConf;
 import tachyon.master.LocalTachyonCluster;
 
 /**
  * Unit tests for <code>tachyon.client.RemoteBlockInStream</code>.
  */
+@RunWith(Parameterized.class)
 public class RemoteBlockInStreamTest {
   private static final int MIN_LEN = 0;
   private static final int MAX_LEN = 255;
   private static final int DELTA = 33;
 
-  private static LocalTachyonCluster mLocalTachyonCluster = null;
-  private static TachyonFS mTfs = null;
+  private LocalTachyonCluster mLocalTachyonCluster = null;
+  private TachyonFS mTfs = null;
+  private String mDataServerClass;
+  private String mRemoteReaderClass;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    // creates a new instance of RemoteBlockInStreamTest for each network type
+    List<Object[]> list = new ArrayList<Object[]>();
+    list.add(new Object[] { new String[] { "tachyon.worker.netty.NettyDataServer",
+        "tachyon.client.tcp.TCPRemoteBlockReader" } });
+    list.add(new Object[] { new String[] { "tachyon.worker.nio.NIODataServer",
+        "tachyon.client.tcp.TCPRemoteBlockReader" } });
+    return list;
+  }
+
+  public RemoteBlockInStreamTest(String[] classes) {
+    mDataServerClass = classes[0];
+    mRemoteReaderClass = classes[1];
+  }
 
   @Rule
   public ExpectedException thrown = ExpectedException.none();
 
-  @AfterClass
-  public static final void afterClass() throws Exception {
+  @After
+  public final void after() throws Exception {
```

Comment:
Does this apply to all tests? Since there are other tests using BeforeClass and AfterClass to speedup unit tests.